packages/framework/immutable-merge/src/Merge.ts

```diff
@@ -51,76 +128,78 @@ export type IMergeRecursionHandler = (options: IMergeOptions, ...objs: any[]) =>
  * be `{}` - no recursion, `{ depth: -1 }` - recurse infinitely
  * @param objs - an array of objects to merge together
  */
-export function immutableMergeCore(options: IMergeOptions, ...objs: any[]): any {
-  const setToMerge = objs.filter(v => _isNonArrayObject(v, true));
-  const { depth = 0, recurse } = options;
-  const processSingle = options.processSingles && setToMerge.length === 1;
+function immutableMergeWorker(mergeOptions: RecursionOption | MergeOptions, singleMode: boolean, ...objs: any[]): any {
+  const setToMerge = objs.filter(v => v && getEntityType(v) === 'object' && Object.getOwnPropertyNames(v).length > 0);
```

Comment:
Why do you need the `getOwnPropertyNames().length` check? It seems like it could be problematic depending on how an object is constructed.

Reply:
The goal is to omit empty objects.  Basically if you call immutableMerge({}, a, {}, {}) it should return a, not { ...a }.  Do you think Object.keys().length would be better?

Reply:
Given how the Object.assign polyfill, maybe the problems I'm thinking about are exceedingly rare. I think this is fine.